File: benchmark/comparison/tasks/09/solution.ts
function gcd(a: number, b: number): number {
  a = Math.abs(a);
  b = Math.abs(b);
  while (b !== 0) {
    const t = b;
    b = a % t;
    a = t;
  }
  return a;
}

console.log(gcd(8, 12));
console.log(gcd(50, 75));
console.log(gcd(7, 13));
console.log(gcd(0, 5));
console.log(gcd(10, 10));

File: benchmark/comparison/tasks/09/solution.py
def gcd(a, b):
    a, b = abs(a), abs(b)
    while b:
        a, b = b, a % b
    return a

print(gcd(12, 8))
print(gcd(100, 75))
print(gcd(7, 13))
print(gcd(0, 5))
print(gcd(10, 0))
